<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="MartinGarlovsky" />

<meta name="date" content="2023-03-02" />

<title>SNP_clusters</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mito_age_fert</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/MartinGarlovsky/mito_age_fert">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SNP_clusters</h1>
<h4 class="author">MartinGarlovsky</h4>
<h4 class="date">2023-03-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mito_age_fert/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230213code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230213)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230213code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230213)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMartinGarlovskymitoageferttreeef4cc07044ad2c04269d86a60f82229755916c8btargetblankef4cc07a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MartinGarlovsky/mito_age_fert/tree/ef4cc07044ad2c04269d86a60f82229755916c8b" target="_blank">ef4cc07</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMartinGarlovskymitoageferttreeef4cc07044ad2c04269d86a60f82229755916c8btargetblankef4cc07a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MartinGarlovsky/mito_age_fert/tree/ef4cc07044ad2c04269d86a60f82229755916c8b" target="_blank">ef4cc07</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  README.html
    Untracked:  check_lines.R
    Untracked:  code/data_wrangling.R
    Untracked:  code/plotting_Script.R
    Untracked:  data/Data_raw_emmely.csv
    Untracked:  data/defence.csv
    Untracked:  data/male_fertility.csv
    Untracked:  data/mito_34sigdiffSNPs_consensus_incl_colnames.csv
    Untracked:  data/mito_mt_copy_number.xlsx
    Untracked:  data/mito_mt_seq_major_alleles_sig_snptable.csv
    Untracked:  data/mito_mt_seq_sig_annotated.csv
    Untracked:  data/mito_mt_seq_sig_annotated.vcf
    Untracked:  data/offence.csv
    Untracked:  data/rawdata_PCA.csv
    Untracked:  data/snp-gene.txt
    Untracked:  data/sperm_metabolic_rate.csv
    Untracked:  data/sperm_viability.csv
    Untracked:  data/wrangled/
    Untracked:  figures/
    Untracked:  output/SNP_clusters.csv
    Untracked:  output/bod_brm.rds
    Untracked:  output/female_rate_dredge.rds
    Untracked:  output/female_rates_bb.Rdata
    Untracked:  output/female_rates_boot.Rdata
    Untracked:  output/female_rates_poly.Rdata
    Untracked:  output/male_fec_dredge.rds
    Untracked:  output/male_hatch_dredge.rds
    Untracked:  output/male_hatch_dredge_reduced.rds
    Untracked:  output/sperm_met_dredge.rds
    Untracked:  output/viab_ctrl_dredge.rds
    Untracked:  output/viab_trt_dredge.rds
    Untracked:  snp_matrix_dobler.csv

Unstaged changes:
    Deleted:    analysis/sperm_comp.Rmd
    Modified:   data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/SNP_clusters.Rmd</code>) and HTML
(<code>docs/SNP_clusters.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MartinGarlovsky/mito_age_fert/blob/ef4cc07044ad2c04269d86a60f82229755916c8b/analysis/SNP_clusters.Rmd" target="_blank">ef4cc07</a>
</td>
<td>
MartinGarlovsky
</td>
<td>
2024-10-10
</td>
<td>
wflow_publish("analysis/SNP_clusters.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(ComplexHeatmap)</code></pre>
<pre><code>Loading required package: grid
========================================
ComplexHeatmap version 2.20.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================</code></pre>
<pre class="r"><code>library(kableExtra)</code></pre>
<pre><code>
Attaching package: &#39;kableExtra&#39;

The following object is masked from &#39;package:dplyr&#39;:

    group_rows</code></pre>
<pre class="r"><code>library(knitrhooks) # install with devtools::install_github(&quot;nathaneastwood/knitrhooks&quot;)</code></pre>
<pre><code>Loading required package: knitr</code></pre>
<pre class="r"><code>library(showtext)</code></pre>
<pre><code>Loading required package: sysfonts
Loading required package: showtextdb</code></pre>
<pre class="r"><code># 
# library(conflicted)

select &lt;- dplyr::select
filter &lt;- dplyr::filter

output_max_height() # a knitrhook option

options(stringsAsFactors = FALSE)

# colour palettes
cbPalette &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#CC79A7&quot;, &quot;#D55E00&quot;, &quot;#0072B2&quot;, &quot;#CC79A7&quot;)
v.pal &lt;- viridis::viridis(n = 3, direction = -1)</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<pre class="r"><code>snp_dat &lt;- read.csv(&#39;data/rawdata_PCA.csv&#39;) %&gt;% 
  select(mito, nuclear, mtn, mtgrp, LINE = line, ND2_1154:ND1_12132) %&gt;% 
  distinct(LINE, .keep_all = TRUE) %&gt;% 
  mutate(across(ND2_1154:ND1_12132, ~ as.numeric(as.factor(.x)) - 1))

colnames(snp_dat)[6:39]</code></pre>
<pre> [1] "ND2_1154"      "CoI_1512"      "CoI_2262"      "CoI_2661"     
 [5] "CoII_3385"     "CoII_3583"     "ATPase6_4247"  "ATPase6_4616" 
 [9] "CoIII_5396"    "tRNA_Glu_6308" "ND5_6989"      "ND5_7232"     
[13] "ND5_7871"      "ND4_8876"      "ND4_8982"      "ND4_9065"     
[17] "ND6_10226"     "ND1_12345"     "ND1_12381"     "LnRNA_13561"  
[21] "Ori_14917"     "Ori_14929"     "Ori_15234"     "Ori_15243"    
[25] "Ori_15321"     "Ori_15322"     "Ori_15610"     "Ori_17255"    
[29] "Ori_18156"     "Ori_19482"     "CoI_1836"      "LnRNA_13934"  
[33] "Ori_15149"     "ND1_12132"    
</pre>
<pre class="r"><code>loc_info &lt;- data.frame(coln = colnames(snp_dat)[6:39]) %&gt;% 
  separate(coln, into = c(&quot;gene&quot;, &quot;position&quot;), remove = FALSE)</code></pre>
<pre><code>Warning: Expected 2 pieces. Additional pieces discarded in 1 rows [10].</code></pre>
<pre class="r"><code>loc_info[10, 2] &lt;- &quot;tRNA_Glu&quot;
loc_info[10, 3] &lt;- &quot;6308&quot;

snp_new &lt;- read.csv(&#39;data/mito_34sigdiffSNPs_consensus_incl_colnames.csv&#39;, header = TRUE) %&gt;% 
  separate(LINE, into = c(&quot;mito&quot;, &quot;nuclear&quot;), sep = &quot;(?&lt;=.)&quot;, remove = FALSE) %&gt;% 
  mutate(mtn = paste0(mito, nuclear))</code></pre>
<pre><code>Warning: Expected 2 pieces. Additional pieces discarded in 27 rows [1, 2, 3, 4, 5, 6, 7,
8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
<pre class="r"><code>cnames &lt;- data.frame(position = gsub(&quot;X&quot;, &quot;&quot;, colnames(snp_new)[4:37])) %&gt;% 
  left_join(loc_info, by = &quot;position&quot;)

colnames(snp_new)[4:37] &lt;- cnames$coln

snp_num &lt;- snp_new %&gt;% mutate(across(ND2_1154:Ori_19482, ~ as.numeric(as.factor(.x)) - 1))

# gene info
snp_gene &lt;- read.delim(&quot;data/snp-gene.txt&quot;)

colnames(snp_num[4:37]) %&gt;% as_tibble() %&gt;% 
  left_join(snp_gene %&gt;% mutate(position = as.character(position)), 
            by = c(&quot;value&quot; = &quot;position&quot;))</code></pre>
<pre># A tibble: 34 × 8
   value          FST gene  substitution maj_A AA_A  maj_B AA_B 
   <chr>        <dbl> <chr> <chr>        <chr> <chr> <chr> <chr>
 1 ND2_1154        NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 2 CoI_1512        NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 3 CoI_1836        NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 4 CoI_2262        NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 5 CoI_2661        NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 6 CoII_3385       NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 7 CoII_3583       NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 8 ATPase6_4247    NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
 9 ATPase6_4616    NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
10 CoIII_5396      NA <NA>  <NA>         <NA>  <NA>  <NA>  <NA> 
# ℹ 24 more rows
</pre>
</div>
<div id="mito-snp-pca" class="section level1">
<h1>mito-SNP PCA</h1>
<p>Using the SNP information from Vaught et al. to identify clusters of
SNPs differentiating between lines. We recoded SNPS with an arbitrary
binary unit and used PCA to identify meaningful groups…</p>
<pre class="r"><code># snp_dat &lt;- femfert %&gt;% 
#   select(mito, nuclear, mtn, mtgrp, LINE, ND2_1154:ND1_12132) %&gt;% 
#   distinct(LINE, .keep_all = TRUE) %&gt;% 
#   mutate(across(ND2_1154:ND1_12132, ~ as.numeric(as.factor(.x)) - 1)) #%&gt;% 
#   # remove SNPs with intermediate frequencies (coded AT) 
#   #select(-Ori_15234, -Ori_15243)

mito_labs &lt;- rowAnnotation(mito = snp_dat$mito,
                           mtgp = snp_dat$mtgrp,
                           nucl = snp_dat$nuclear,
                           LINE = snp_dat$LINE,
                           col = list(mito = setNames(v.pal, unique(snp_dat$mito)),
                                      mtgp = setNames(rainbow(9), unique(snp_dat$mtgrp)),
                                      nucl = setNames(v.pal, unique(snp_dat$nuclear)),
                                      LINE = setNames(rainbow(27), unique(snp_dat$LINE))),
                           title = NULL
                           )

mito_labs &lt;- rowAnnotation(mito = snp_new$mito,
                           nucl = snp_new$nuclear,
                           LINE = snp_new$LINE,
                           col = list(mito = setNames(v.pal, unique(snp_new$mito)),
                                      nucl = setNames(v.pal, unique(snp_new$nuclear)),
                                      LINE = setNames(rainbow(27), unique(snp_new$LINE))),
                           title = NULL
                           )

#pdf(&#39;figures/SNP_heatmap.pdf&#39;, height = 8, width = 5)
Heatmap(as.matrix(snp_new %&gt;% select(ND2_1154:Ori_19482)), 
        #col = c(&#39;black&#39;, &#39;grey&#39;),
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        left_annotation = mito_labs,
        #top_annotation = haha, 
        show_row_names = FALSE, 
        show_column_names = FALSE,
        #row_split = 2, 
        #column_split = 3,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL)</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()

pca.cor &lt;- prcomp(snp_num %&gt;% select(ND2_1154:Ori_19482))

broom::tidy(pca.cor, &#39;pcs&#39;) %&gt;% 
  mutate(percent = percent * 100, cumulative = cumulative * 100)</code></pre>
<pre># A tibble: 27 × 4
      PC  std.dev percent cumulative
   <dbl>    <dbl>   <dbl>      <dbl>
 1     1 2.45e+ 0  82.8         82.8
 2     2 8.15e- 1   9.19        92.0
 3     3 4.57e- 1   2.89        94.9
 4     4 4.02e- 1   2.23        97.1
 5     5 3.65e- 1   1.84        99.0
 6     6 2.16e- 1   0.642       99.6
 7     7 1.65e- 1   0.375      100  
 8     8 5.12e-16   0          100  
 9     9 2.43e-16   0          100  
10    10 2.43e-16   0          100  
# ℹ 17 more rows
</pre>
<pre class="r"><code># biplot(pca.cor)
# biplot(pca.cov)

# screeplot(pca.cor)
# screeplot(pca.cov)

PCA_dat &lt;- broom::augment(pca.cor, data = snp_num)

PCA_dat %&gt;%
  ggplot(aes(x = .fittedPC1, y = .fittedPC2, colour = mtn)) +
  geom_point(size = 8, alpha = .5) +
  scale_colour_viridis_d() +
  labs(x = paste0(&#39;PC1 (&#39;, (100*round(summary(pca.cor)$importance[2, 1], 3)), &#39;%)&#39;),
       y = paste0(&#39;PC2 (&#39;, (100*round(summary(pca.cor)$importance[2, 2], 3)), &#39;%)&#39;)) +
  theme_bw() +
  theme(legend.position = &#39;&#39;,
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
        axis.text = element_text(size = 10)) +
  ggrepel::geom_text_repel(aes(label = LINE), colour = &#39;black&#39;, max.overlaps = 20) +
  #ggsave(&#39;plots/PCA_12.pdf&#39;, height = 3.4, width = 4.5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<pre><code>Warning: The `legend.text.align` argument of `theme()` is deprecated as of ggplot2
3.5.0.
ℹ Please use theme(legend.text = element_text(hjust)) instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>bind_rows(PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC1, PCb = .fittedPC2),
          PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC1, PCb = .fittedPC3),
          PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC2, PCb = .fittedPC3)) %&gt;% 
  mutate(pc = rep(c(&#39;PC1 (81.2%) vs. PC2 (8.6%)&#39;,
                    &#39;PC1 (81.2%) vs. PC3 (3.1%)&#39;,
                    &#39;PC2 (8.6%) vs. PC3 (3.1%)&#39;), each = nrow(.)/3)) %&gt;%
  ggplot(aes(x = PCa, y = PCb, colour = mito)) +
  geom_point(size = 8, alpha = .5) +
  scale_colour_viridis_d() +
  facet_wrap(~pc) +
  theme_bw() +
  theme(legend.position = &#39;&#39;,
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
        axis.text = element_text(size = 10),
        axis.title = element_blank(),
        strip.text = element_text(size = 15)) +
  ggrepel::geom_text_repel(aes(label = LINE), colour = &#39;black&#39;, max.overlaps = 20) +
  #ggsave(&#39;plots/PCA_12.pdf&#39;, height = 3.4, width = 4.5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># which SNPs are contributing most to differences in PCs?
data.frame(pca.cor$rotation) %&gt;% rownames_to_column(var = &quot;gene&quot;) %&gt;% 
  ggplot(aes(x = gene, y = PC1)) +
  geom_point() +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># library(factoextra)
# library(cluster)
# 
# fviz_nbclust(snp_dat %&gt;% select(ND2_1154:ND1_12132), FUN = hcut)
# fviz_nbclust(snp_dat %&gt;% select(ND2_1154:ND1_12132), FUN = hcut, method = &quot;wss&quot;)
# fviz_nbclust(snp_dat %&gt;% select(ND2_1154:ND1_12132), FUN = hcut, method = &quot;silhouette&quot;)
# 
# fviz_nbclust(snp_dat %&gt;% select(ND2_1154:ND1_12132), kmeans, nstart = 25, k.max = 9,
#              method = &quot;gap_stat&quot;, nboot = 1000) +
#   labs(subtitle = &quot;Gap statistic method&quot;)

# Dissimilarity matrix
d &lt;- dist(snp_num %&gt;% select(contains(&quot;_&quot;)), method = &quot;euclidean&quot;)

# Hierarchical clustering using Complete Linkage
hc1 &lt;- hclust(d, method = &quot;complete&quot;)

# Plot the obtained dendrogram
#plot(hc1, cex = 0.6, hang = -1)

# Cut tree into 2 groups
mito_cluster &lt;- cutree(hc1, k = 2)

# Number of members in each cluster
table(mito_cluster)</code></pre>
<pre>mito_cluster
 1  2 
 8 19 
</pre>
<pre class="r"><code># or... manually based on observing PCs...
snp_dat2 &lt;- snp_new %&gt;% 
  mutate(mito2 = case_when(grepl(&quot;B&quot;, x = mito) ~ &quot;b_like&quot;,
                           LINE == &quot;CA1&quot;| LINE == &quot;CA3&quot; ~ &quot;b_like&quot;,
                           grepl(&quot;C&quot;, x = mito) ~ &quot;c_like&quot;,
                           LINE == &quot;AA3&quot; ~ &quot;c_like&quot;,
                           TRUE ~ &quot;a_like&quot;))

mito_labs2 &lt;- rowAnnotation(mito = snp_new$mito,
                            #mtgp = snp_dat$mtgrp,
                            nucl = snp_new$nuclear,
                            mtcl = snp_dat2$mito2,
                            col = list(mito = setNames(rev(v.pal), unique(snp_new$mito)),
                                       #mtgp = setNames(rainbow(9), unique(snp_dat$mtgrp)),
                                       nucl = setNames(rev(v.pal), unique(snp_new$nuclear)),
                                       mtcl = setNames(rainbow(3), unique(snp_dat2$mito2))),
                            title = NULL
                            )

#pdf(&#39;figures/SNP_heatmap2.pdf&#39;, height = 8, width = 5)
Heatmap(as.matrix(snp_num %&gt;% select(ND2_1154:Ori_19482)), 
        col = c(&#39;black&#39;, &#39;grey&#39;),
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        left_annotation = mito_labs2,
        #top_annotation = haha, 
        show_row_names = FALSE, 
        show_column_names = FALSE,
        row_split = 2, 
        #column_split = 3,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL)</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()

# Heatmap(as.matrix(snp_dat %&gt;% select(ND2_1154:ND1_12132)), 
#         col = cbPalette[4:8],
#         heatmap_legend_param = list(title = &quot;SNP&quot;),
#         left_annotation = mito_labs2,
#         #top_annotation = haha, 
#         show_row_names = FALSE, 
#         show_column_names = FALSE,
#         row_split = 2, 
#         #column_split = 3,
#         column_gap = unit(0, &quot;mm&quot;),
#         row_title = NULL,
#         column_title = NULL,
#         cell_fun = function(j, i, x, y, w, h, col) { # add text to each grid
#         grid.text(mat_snps[i, j], x, y)})

# SNP graph
snpd2 &lt;- read.csv(&#39;data/rawdata_PCA.csv&#39;) %&gt;% 
  select(mito, nuclear, mtn, mtgrp, LINE = line, ND2_1154:ND1_12132) %&gt;% 
  distinct(LINE, .keep_all = TRUE)

mat_snps &lt;- as.matrix(snpd2 %&gt;% select(ND2_1154:ND1_12132))


mat_snps &lt;- as.matrix(snp_dat2 %&gt;% select(ND2_1154:Ori_19482))

#pdf(&#39;figures/SNP_map.pdf&#39;, height = 5.5, width = 8)
Heatmap(as.matrix(snp_new %&gt;% select(ND2_1154:Ori_19482)), 
        col = cbPalette[4:7], # col = c(cbPalette[4], cbPalette[1], cbPalette[5:7]), # to plot old version
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        row_order = order(snp_dat2$mito2),
        left_annotation = mito_labs2,
        #top_annotation = haha, 
        show_row_names = FALSE, 
        #show_column_names = FALSE,
        #row_split = 2, 
        #column_split = 3,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL,
        cell_fun = function(j, i, x, y, w, h, col) { # add text to each grid
        grid.text(mat_snps[i, j], x, y)
    })</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-3-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()

snp_new$LnRNA_13934</code></pre>
<pre> [1] "T" "T" "T" "T" "T" "T" "T" "T" "T" "T" "C" "C" "C" "T" "C" "C" "C" "C" "C"
[20] "T" "C" "T" "T" "T" "T" "T" "T"
</pre>
<pre class="r"><code>snpd2$LnRNA_13934</code></pre>
<pre> [1] "T" "T" "T" "T" "A" "A" "A" "T" "T" "T" "T" "T" "A" "T" "A" "T" "T" "T" "T"
[20] "T" "T" "A" "A" "A" "A" "T" "T"
</pre>
<pre class="r"><code>data.frame(LINE = snpd2$LINE,
           LnRNA_13934 = snpd2$LnRNA_13934) %&gt;% 
  left_join(data.frame(LINE = snp_new$LINE,
                       LnRNA_13934 = snp_new$LnRNA_13934),
            by = &quot;LINE&quot;)</code></pre>
<pre>   LINE LnRNA_13934.x LnRNA_13934.y
1   AA1             T             T
2   AB1             T             T
3   AC1             T             T
4   BA1             T             T
5   BB1             A             C
6   BC1             A             C
7   CA1             A             C
8   CB1             T             T
9   CC1             T             T
10  AA2             T             T
11  AB2             T             T
12  AC2             T             T
13  BA2             A             C
14  BB2             T             T
15  BC2             A             C
16  CA2             T             T
17  CB2             T             T
18  CC2             T             T
19  AA3             T             T
20  AB3             T             T
21  AC3             T             T
22  BA3             A             C
23  BB3             A             C
24  BC3             A             C
25  CA3             A             C
26  CB3             T             T
27  CC3             T             T
</pre>
</div>
<div id="new-analysis" class="section level1">
<h1>new analysis</h1>
<pre class="r"><code>snp_susi &lt;- read.csv(&#39;data/mito_mt_seq_major_alleles_sig_snptable.csv&#39;, header = TRUE)

colnames(snp_susi)[1] &lt;- &quot;LINE&quot;
colnames(snp_susi)[2:45] &lt;- gsub(&quot;X&quot;, &quot;&quot;, x = colnames(snp_susi)[2:45])

snp_susi &lt;- snp_susi %&gt;% 
  separate(LINE, into = c(&quot;mito&quot;, &quot;nuclear&quot;), sep = &quot;(?&lt;=.)&quot;, remove = FALSE) %&gt;% 
  mutate(mtn = paste0(mito, nuclear))</code></pre>
<pre><code>Warning: Expected 2 pieces. Additional pieces discarded in 27 rows [1, 2, 3, 4, 5, 6, 7,
8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
<pre class="r"><code>snp_num &lt;- snp_susi %&gt;% mutate(across(`1154`:`17400`, ~ as.numeric(as.factor(.x)) - 1))


mito_labs &lt;- rowAnnotation(mito = snp_susi$mito,
                           nucl = snp_susi$nuclear,
                           LINE = snp_susi$LINE,
                           col = list(mito = setNames(rev(v.pal), unique(snp_susi$mito)),
                                      nucl = setNames(rev(v.pal), unique(snp_susi$nuclear)),
                                      LINE = setNames(rainbow(27), unique(snp_susi$LINE))),
                           title = NULL
                           )

#pdf(&#39;figures/SNP_heatmap.pdf&#39;, height = 8, width = 5)
Heatmap(as.matrix(snp_susi %&gt;% select(`1154`:`17400`)), 
        #col = c(&#39;black&#39;, &#39;grey&#39;),
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        left_annotation = mito_labs,
        show_row_names = FALSE, 
        show_column_names = FALSE,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL)</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()

pca.cor &lt;- prcomp(snp_num %&gt;% select(`1154`:`17400`))

broom::tidy(pca.cor, &#39;pcs&#39;) %&gt;% 
  mutate(percent = percent * 100, cumulative = cumulative * 100)</code></pre>
<pre># A tibble: 27 × 4
      PC  std.dev percent cumulative
   <dbl>    <dbl>   <dbl>      <dbl>
 1     1 2.30e+ 0  77.3         77.3
 2     2 8.19e- 1   9.85        87.2
 3     3 6.36e- 1   5.94        93.1
 4     4 5.50e- 1   4.43        97.6
 5     5 2.36e- 1   0.819       98.4
 6     6 2.24e- 1   0.734       99.1
 7     7 2.07e- 1   0.627       99.7
 8     8 1.35e- 1   0.267      100  
 9     9 6.50e-16   0          100  
10    10 2.26e-16   0          100  
# ℹ 17 more rows
</pre>
<pre class="r"><code># biplot(pca.cor)
# biplot(pca.cov)

# screeplot(pca.cor)
# screeplot(pca.cov)

PCA_dat &lt;- broom::augment(pca.cor, data = snp_num)

PCA_dat %&gt;%
  ggplot(aes(x = .fittedPC1, y = .fittedPC2, fill = mtn)) +
  geom_point(size = 8, alpha = .5, pch = 21) +
  scale_fill_viridis_d() +
  labs(x = paste0(&#39;PC1 (&#39;, (100*round(summary(pca.cor)$importance[2, 1], 3)), &#39;%)&#39;),
       y = paste0(&#39;PC2 (&#39;, (100*round(summary(pca.cor)$importance[2, 2], 3)), &#39;%)&#39;)) +
  coord_cartesian(ylim = c(-1, 1)) +
  theme_bw() +
  theme(legend.position = &#39;&#39;,
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
        axis.text = element_text(size = 10)) +
  ggrepel::geom_text_repel(aes(label = LINE), colour = &#39;black&#39;, max.overlaps = 20) +
  #ggsave(&#39;plots/PCA_12.pdf&#39;, height = 3.4, width = 4.5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>bind_rows(PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC1, PCb = .fittedPC2),
          PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC1, PCb = .fittedPC3),
          PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC2, PCb = .fittedPC3)) %&gt;% 
  mutate(pc = rep(c(&#39;PC1 (77.3%) vs. PC2 (9.85%)&#39;,
                    &#39;PC1 (77.3%) vs. PC3 (5.94%)&#39;,
                    &#39;PC2 (9.85%) vs. PC3 (5.94%)&#39;), each = nrow(.)/3)) %&gt;%
  ggplot(aes(x = PCa, y = PCb, colour = mito)) +
  geom_point(size = 8, alpha = .5) +
  scale_colour_viridis_d() +
  coord_cartesian(ylim = c(-1, 1)) +
  facet_wrap(~pc) +
  theme_bw() +
  theme(legend.position = &#39;&#39;,
        legend.text.align = 0,
        legend.text = element_text(size = 12),
        legend.background = element_blank(),
        axis.text = element_text(size = 10),
        axis.title = element_blank(),
        strip.text = element_text(size = 15)) +
  ggrepel::geom_text_repel(aes(label = LINE), colour = &#39;black&#39;, max.overlaps = 20) +
  #ggsave(&#39;plots/PCA_12.pdf&#39;, height = 3.4, width = 4.5, dpi = 600, useDingbats = FALSE) +
  NULL</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># which SNPs are contributing most to differences in PCs?
data.frame(pca.cor$rotation) %&gt;% rownames_to_column(var = &quot;gene&quot;) %&gt;% 
  ggplot(aes(x = gene, y = PC1)) +
  geom_point() +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(factoextra)</code></pre>
<pre><code>Welcome! Want to learn more? See two factoextra-related books at https://goo.gl/ve3WBa</code></pre>
<pre class="r"><code>library(cluster)

fviz_nbclust(snp_num %&gt;% select(`1154`:`17400`), FUN = hcut)</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fviz_nbclust(snp_num %&gt;% select(`1154`:`17400`), FUN = hcut, method = &quot;wss&quot;)</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fviz_nbclust(snp_num %&gt;% select(`1154`:`17400`), FUN = hcut, method = &quot;silhouette&quot;)</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># fviz_nbclust(snp_num %&gt;% select(`1154`:`17400`), kmeans, nstart = 25, k.max = 9,
#              method = &quot;gap_stat&quot;, nboot = 1000) +
#   labs(subtitle = &quot;Gap statistic method&quot;)

# Dissimilarity matrix
d &lt;- dist(snp_num %&gt;% select(`1154`:`17400`), method = &quot;euclidean&quot;)

# Hierarchical clustering using Complete Linkage
hc1 &lt;- hclust(d, method = &quot;complete&quot;)

# Plot the obtained dendrogram
#plot(hc1, cex = 0.6, hang = -1)

# Cut tree into 2 groups
mito_cluster &lt;- cutree(hc1, k = 9)

# Number of members in each cluster
table(mito_cluster)</code></pre>
<pre>mito_cluster
1 2 3 4 5 6 7 8 9 
4 1 2 1 1 2 9 5 2 
</pre>
<pre class="r"><code>data.frame(LINE = snp_susi$LINE,
           mito_cluster)</code></pre>
<pre>   LINE mito_cluster
1   AA1            1
2   AA2            1
3   AA3            2
4   AB1            3
5   AB2            1
6   AB3            4
7   AC1            1
8   AC2            5
9   AC3            3
10  BA1            6
11  BA2            7
12  BA3            7
13  BB1            7
14  BB2            6
15  BB3            7
16  BC1            7
17  BC2            7
18  BC3            7
19  CA1            7
20  CA2            8
21  CA3            7
22  CB1            8
23  CB2            9
24  CB3            8
25  CC1            8
26  CC2            8
27  CC3            9
</pre>
<pre class="r"><code># or... manually based on observing PCs...
snp_dat2 &lt;- snp_susi %&gt;% 
  mutate(mito2 = case_when(grepl(&quot;B&quot;, x = mito) ~ &quot;b_like&quot;,
                           LINE == &quot;CA1&quot;| LINE == &quot;CA3&quot; ~ &quot;b_like&quot;,
                           grepl(&quot;C&quot;, x = mito) ~ &quot;c_like&quot;,
                           LINE == &quot;AA3&quot; ~ &quot;AA3&quot;,
                           TRUE ~ &quot;a_like&quot;),
         mito_snp = mito_cluster)

#snp_dat2 %&gt;% write_csv(&quot;output/SNP_clusters.csv&quot;)

#snp_dat2 &lt;- snp_dat2 %&gt;% arrange(nuclear, )

snp_annot &lt;- read.csv(&quot;data/mito_mt_seq_sig_annotated.csv&quot;)

Archambault.pal &lt;- MetBrewer::met.brewer(&#39;Archambault&#39;)

vmha &lt;- HeatmapAnnotation(gene = snp_annot$gene, 
                          substitution = snp_annot$substitution,
                          # anno_block(gp = gpar(fill = viridis::viridis(n = 15, option = &quot;H&quot;)),
                          #            labels = snp_annot$gene, 
                          #            labels_gp = gpar(col = &quot;white&quot;, fontsize = 10)),
                          col = list(gene = setNames(viridis::viridis(n = 15, option = &quot;H&quot;),
                                                       unique(snp_annot$gene)),
                                     substitution = setNames(c(Archambault.pal[c(1, 4, 7)]),
                                                       unique(snp_annot$substitution))))

mat_snps &lt;- as.matrix(snp_dat2 %&gt;% select(`1154`:`17400`))

mito_labs2 &lt;- rowAnnotation(msnp = snp_dat2$mito_snp,
                            mito = snp_dat2$mito,
                            nucl = snp_dat2$nuclear,
                            #mtcl = snp_dat2$mito2,
                            col = list(mito = setNames(v.pal, unique(snp_dat2$mito)),
                                       msnp = setNames(viridis::viridis(n = 9, option = &quot;H&quot;),
                                                       unique(snp_dat2$mito_snp)),
                                       nucl = setNames(v.pal, unique(snp_dat2$nuclear)),
                                       mtcl = setNames(rainbow(4), unique(snp_dat2$mito2))),
                            title = NULL
                            )

snp_mat_n &lt;- as.matrix(snp_num %&gt;% select(`1154`:`17400`))
rownames(snp_mat_n) &lt;- snp_num$LINE

#pdf(&#39;figures/SNP_tree_may.pdf&#39;, height = 5.5, width = 12)
Heatmap(snp_mat_n, 
        col = c(&#39;grey50&#39;, &#39;grey&#39;),
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        left_annotation = mito_labs2,
        top_annotation = vmha, 
        show_row_names = TRUE, 
        show_column_names = TRUE,
        #row_split = 2, 
        #column_split = 3,
        cluster_columns = FALSE,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL,
        cell_fun = function(j, i, x, y, w, h, col) { # add text to each grid
          grid.text(mat_snps[i, j], x, y)
          }
)</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()



snp_mat_numbs &lt;- matrix(as.numeric(as.factor(as.matrix(snp_susi %&gt;% select(`1154`:`17400`)))), ncol = 44)
colnames(snp_mat_numbs) &lt;- colnames(snp_susi %&gt;% select(`1154`:`17400`))
rownames(snp_mat_numbs) &lt;- snp_susi$LINE

Heatmap(snp_mat_numbs, 
        col = cbPalette[4:7], 
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        row_order = order(snp_dat2$mito_snp),
        left_annotation = mito_labs2,
        top_annotation = vmha, 
        show_row_names = TRUE, 
        #show_column_names = FALSE,
        #row_split = 2, 
        #column_split = 3,
        cluster_columns = FALSE,
        column_gap = unit(0, &quot;mm&quot;),
        rect_gp = gpar(col = &quot;white&quot;, lwd = 2),
        row_title = NULL,
        column_title = NULL,
        cell_fun = function(j, i, x, y, w, h, col) { # add text to each grid
          grid.text(mat_snps[i, j], x, y)
          }
        )</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>HeatmapAnnotation(foo = anno_block(gp = gpar(fill = 2:4),
                                   labels = c(&quot;group1&quot;, &quot;group2&quot;, &quot;group3&quot;), 
                                   labels_gp = gpar(col = &quot;white&quot;, fontsize = 10)))</code></pre>
<pre>A HeatmapAnnotation object with 1 annotation
  name: heatmap_annotation_6 
  position: column 
  items: unknown 
  width: 1npc 
  height: 7.53294444444444mm 
  this object is subsettable

 name annotation_type color_mapping             height
  foo    anno_block()               7.53294444444444mm
</pre>
<pre class="r"><code>snp_matrix &lt;- as.matrix(snp_susi %&gt;% select(`1154`:`17400`))
rownames(snp_matrix) &lt;- snp_susi$LINE

#pdf(&#39;figures/SNP_map_May.pdf&#39;, height = 5.5, width = 12)
Heatmap(snp_matrix, 
        col = cbPalette[4:7], # col = c(cbPalette[4], cbPalette[1], cbPalette[5:7]), # to plot old version
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        #row_order = order(snp_dat2$mito_snp),
        left_annotation = mito_labs2,
        top_annotation = vmha, 
        show_row_names = TRUE, 
        #show_column_names = FALSE,
        #row_split = 2, 
        #column_split = 3,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL,
        cell_fun = function(j, i, x, y, w, h, col) { # add text to each grid
          grid.text(mat_snps[i, j], x, y)
          }
        )</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-4-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()


# bind_rows(PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC1, PCb = .fittedPC2),
#           PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC1, PCb = .fittedPC3),
#           PCA_dat %&gt;% select(mito, nuclear, LINE, PCa = .fittedPC2, PCb = .fittedPC3)) %&gt;% 
#   mutate(pc = rep(c(&#39;PC1 (77.3%) vs. PC2 (9.85%)&#39;,
#                     &#39;PC1 (77.3%) vs. PC3 (5.94%)&#39;,
#                     &#39;PC2 (9.85%) vs. PC3 (5.94%)&#39;), each = nrow(.)/3)) %&gt;% 
#   left_join(snp_dat2 %&gt;% select(LINE, mito2, mito_cluster)) %&gt;% 
#   mutate(mito_cluster = as.factor(mito_cluster)) %&gt;%
#   ggplot(aes(x = PCa, y = PCb, colour = mito_cluster)) +
#   geom_point(size = 8, alpha = .5) +
#   scale_colour_viridis_d(option = &quot;H&quot;) +
#   scale_fill_viridis_d(option = &quot;H&quot;) +
#   coord_cartesian(ylim = c(-.78, .75)) +
#   guides(colour = guide_legend(override.aes = list(alpha = 1))) +
#   facet_wrap(~pc) +
#   theme_bw() +
#   theme(#legend.position = &#39;&#39;,
#         legend.text.align = 0,
#         legend.text = element_text(size = 12),
#         legend.background = element_blank(),
#         axis.text = element_text(size = 10),
#         axis.title = element_blank(),
#         strip.text = element_text(size = 15)) +
#   ggrepel::geom_text_repel(aes(label = LINE), colour = &#39;black&#39;, max.overlaps = 20) +
#   #ggsave(&#39;plots/PCA_12.pdf&#39;, height = 3.4, width = 4.5, dpi = 600, useDingbats = FALSE) +
#   NULL</code></pre>
</div>
<div id="haplomap" class="section level1">
<h1>haplomap</h1>
<pre class="r"><code>library(geneHapR)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;pegas&#39;:
  method      from    
  [.haplotype genetics</code></pre>
<pre class="r"><code>snp_t &lt;- data.frame(t(snp_susi[, 4:47])) %&gt;% 
  rownames_to_column(var = &quot;POS&quot;)

colnames(snp_t)[2:28] &lt;- snp_susi$LINE

hapdat &lt;- bind_cols(CHROM = &quot;X&quot;,
                    POS = as.numeric(snp_t[, 1]), 
                    REF = t(apply(snp_t[, 2:28], 1, unique))[, 1],
                    ALT = t(apply(snp_t[, 2:28], 1, unique))[, 2],
                    INFO = paste0(&quot;CDS=&quot;, 
                                  t(apply(snp_t[, 2:28], 1, unique))[, 1], 
                                  &quot;&gt;&quot;,
                                  t(apply(snp_t[, 2:28], 1, unique))[, 2]),
                    snp_t[, 2:28])

hapResult &lt;- table2hap(hapdat)

hapSummary &lt;- hap_summary(hapResult)
hapSummary</code></pre>
<pre>
Accssions: 27
Sites: 44
  Indels: NA
  SNPs:   44

Haplotypes: 9 
  H001  9   BA2, BA3, BB1, BB3, BC1, BC2, ...   
  H002  5   CA2, CB1, CB3, CC1, CC2 
  H003  4   AA1, AA2, AB2, AC1  
  H004  2   BA1, BB2    
  H005  2   CB2, CC3    
  H006  2   AB1, AC3    
  H007  1   AA3 
  H008  1   AC2 
  H009  1   AB3 
  
Options:
 hapPrefix: H
 CHROM: X
 POS:   1154-17400
 hetero_remove: YES
 NA_remove: YES

# A tibble: 13 × 47
  Hap    `1154`  `1512`  `1836` `2262` `2661` `3385` `3583` `4247` `4533` `4616`
  <chr>  <chr>   <chr>   <chr>  <chr>  <chr>  <chr>  <chr>  <chr>  <chr>  <chr> 
1 CHR    X       X       X      X      X      X      X      X      X      X     
2 POS    1154    1512    1836   2262   2661   3385   3583   4247   4533   4616  
3 INFO   CDS=T>C CDS=T>C CDS=G… CDS=A… CDS=T… CDS=T… CDS=C… CDS=T… CDS=G… CDS=A…
4 ALLELE T/C     T/C     G/A    A/G    T/C    T/C    C/T    T/C    G/A    A/T   
5 H001   C       C       A      G      C      C      T      C      G      T     
6 H002   C       C       G      G      C      C      T      C      G      T     
7 H003   T       T       G      A      T      T      C      T      G      A     
</pre>
<pre class="r"><code>plotHapTable(hapSummary, 
             angle = 45,
             INFO_tag = &quot;CDSChange&quot;)</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 46, is not a
multiple of vector length 3 of arg 1</code></pre>
<pre><code>Warning: `guide_colourbar()` needs continuous scales.</code></pre>
<pre><code>Warning: Removed 16 rows containing missing values or values outside the scale range
(`geom_text()`).</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hapNet &lt;- get_hapNet(hapSummary,
                     AccINFO = AccINFO,
                     groupName = &quot;Type&quot;)</code></pre>
<pre><code>Warning in getHapGroup(hapSummary, AccINFO = AccINFO, groupName = groupName, :
27 accession(s) not in &#39;AccINFO&#39;, eg.: BA2, BA3, BB1</code></pre>
<pre><code>Type of those accessions will be assigned Unknown</code></pre>
<pre class="r"><code>plotHapNet(hapNet,
           size = &quot;freq&quot;,                   # circle size
           scale = &quot;log2&quot;,                 # scale circle with &#39;log10(size + 1)&#39;
           cex = 0.8,                       # size of hap symbol
           col.link = 1,                    # link colors
           link.width = 1,                  # link widths
           show.mutation = 2,               # mutation types one of c(0,1,2,3)
           legend = FALSE#c(-12.5, 7)
           )</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-designations" class="section level1">
<h1>figure designations!?</h1>
<pre class="r"><code>plot_snp &lt;- data.frame(LINE = snp_dat2$LINE,
           mito_snp = c(&quot;1&quot;, &quot;1&quot;, &quot;3&quot;, 
                        &quot;1&quot;, &quot;1&quot;, &quot;2&quot;,
                        &quot;1&quot;, &quot;2&quot;, &quot;1&quot;,
                        &quot;7&quot;, &quot;8&quot;, &quot;8&quot;,
                        &quot;8&quot;, &quot;7&quot;, &quot;8&quot;,
                        &quot;9&quot;, &quot;9&quot;, &quot;9&quot;, 
                        &quot;9&quot;, &quot;5&quot;, &quot;8&quot;,
                        &quot;6&quot;, &quot;4&quot;, &quot;4&quot;,
                        &quot;4&quot;, &quot;5&quot;, &quot;4&quot;)) #%&gt;% write_csv(&quot;output/SNP_clusters.csv&quot;)

plot_snp</code></pre>
<pre>   LINE mito_snp
1   AA1        1
2   AA2        1
3   AA3        3
4   AB1        1
5   AB2        1
6   AB3        2
7   AC1        1
8   AC2        2
9   AC3        1
10  BA1        7
11  BA2        8
12  BA3        8
13  BB1        8
14  BB2        7
15  BB3        8
16  BC1        9
17  BC2        9
18  BC3        9
19  CA1        9
20  CA2        5
21  CA3        8
22  CB1        6
23  CB2        4
24  CB3        4
25  CC1        4
26  CC2        5
27  CC3        4
</pre>
<pre class="r"><code>snp_susi$LINE == plot_snp$LINE</code></pre>
<pre> [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[16] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
</pre>
<pre class="r"><code>mito_again &lt;- rowAnnotation(mito = snp_susi$mito,
                            nucl = snp_susi$nuclear,
                            mtgp = plot_snp$mito_snp,
                            mtcl = snp_dat2$mito2,
                            col = list(mito = setNames(rev(v.pal), unique(snp_susi$mito)),
                                       mtgp = setNames(viridis::viridis(n = 9, option = &quot;H&quot;)[as.numeric(noquote(unique(plot_snp$mito_snp)))],
                                                       unique(plot_snp$mito_snp)),
                                       nucl = setNames(rev(v.pal), unique(snp_susi$nuclear)),
                                       mtcl = setNames(rainbow(4), unique(snp_dat2$mito2))),
                            title = NULL
                            )

#pdf(&#39;figures/SNP_map_correct_ordered.pdf&#39;, height = 5.5, width = 8)
Heatmap(as.matrix(snp_susi %&gt;% select(`1154`:`17400`)), 
        col = cbPalette[4:7], # col = c(cbPalette[4], cbPalette[1], cbPalette[5:7]), # to plot old version
        heatmap_legend_param = list(title = &quot;SNP&quot;),
        row_order = order(plot_snp$mito_snp),
        left_annotation = mito_again,
        #top_annotation = haha, 
        show_row_names = FALSE, 
        #show_column_names = FALSE,
        #row_split = 2, 
        #column_split = 3,
        column_gap = unit(0, &quot;mm&quot;),
        row_title = NULL,
        column_title = NULL,
        cell_fun = function(j, i, x, y, w, h, col) { # add text to each grid
        grid.text(mat_snps[i, j], x, y)
    })</code></pre>
<p><img src="figure/SNP_clusters.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre>R version 4.4.0 (2024-04-24)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/London
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] geneHapR_1.2.4        cluster_2.1.6         factoextra_1.0.7.999 
 [4] showtext_0.9-7        showtextdb_3.0        sysfonts_0.8.9       
 [7] knitrhooks_0.0.4      knitr_1.48            kableExtra_1.4.0     
[10] ComplexHeatmap_2.20.0 lubridate_1.9.3       forcats_1.0.0        
[13] stringr_1.5.1         dplyr_1.1.4           purrr_1.0.2          
[16] readr_2.1.5           tidyr_1.3.1           tibble_3.2.1         
[19] ggplot2_3.5.1         tidyverse_2.0.0       workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] splines_4.4.0               later_1.3.2                
  [3] BiocIO_1.14.0               bitops_1.0-9               
  [5] XML_3.99-0.17               lifecycle_1.0.4            
  [7] sf_1.0-17                   rstatix_0.7.2              
  [9] doParallel_1.0.17           rprojroot_2.0.4            
 [11] processx_3.8.4              lattice_0.22-6             
 [13] MASS_7.3-61                 backports_1.5.0            
 [15] magrittr_2.0.3              sass_0.4.9                 
 [17] rmarkdown_2.28              vcfR_1.15.0                
 [19] jquerylib_0.1.4             yaml_2.3.10                
 [21] httpuv_1.6.15               DBI_1.2.3                  
 [23] RColorBrewer_1.1-3          maps_3.4.2                 
 [25] abind_1.4-8                 zlibbioc_1.50.0            
 [27] GenomicRanges_1.56.1        BiocGenerics_0.50.0        
 [29] RCurl_1.98-1.16             git2r_0.33.0               
 [31] circlize_0.4.16             GenomeInfoDbData_1.2.12    
 [33] IRanges_2.38.1              S4Vectors_0.42.1           
 [35] grImport_0.9-7              ggrepel_0.9.6              
 [37] gdata_3.0.0                 MetBrewer_0.2.0            
 [39] vegan_2.6-8                 units_0.8-5                
 [41] permute_0.9-7               svglite_2.1.3              
 [43] codetools_0.2-20            DelayedArray_0.30.1        
 [45] xml2_1.3.6                  tidyselect_1.2.1           
 [47] shape_1.4.6.1               UCSC.utils_1.0.0           
 [49] farver_2.1.2                viridis_0.6.5              
 [51] matrixStats_1.4.1           stats4_4.4.0               
 [53] GenomicAlignments_1.40.0    jsonlite_1.8.9             
 [55] GetoptLong_1.0.5            e1071_1.7-16               
 [57] pinfsc50_1.3.0              Formula_1.2-5              
 [59] iterators_1.0.14            systemfonts_1.1.0          
 [61] foreach_1.5.2               tools_4.4.0                
 [63] stringdist_0.9.12           Rcpp_1.0.13                
 [65] glue_1.8.0                  gridExtra_2.3              
 [67] genetics_1.3.8.1.3          SparseArray_1.4.8          
 [69] mgcv_1.9-1                  xfun_0.48                  
 [71] MatrixGenerics_1.16.0       GenomeInfoDb_1.40.1        
 [73] withr_3.0.1                 combinat_0.0-8             
 [75] fastmap_1.2.0               fansi_1.0.6                
 [77] shinyjs_2.1.0               callr_3.7.6                
 [79] digest_0.6.37               mime_0.12                  
 [81] timechange_0.3.0            R6_2.5.1                   
 [83] colorspace_2.1-1            gtools_3.9.5               
 [85] utf8_1.2.4                  generics_0.1.3             
 [87] rtracklayer_1.64.0          class_7.3-22               
 [89] httr_1.4.7                  S4Arrays_1.4.1             
 [91] whisker_0.4.1               pkgconfig_2.0.3            
 [93] gtable_0.3.5                XVector_0.44.0             
 [95] htmltools_0.5.8.1           carData_3.0-5              
 [97] clue_0.3-65                 scales_1.3.0               
 [99] Biobase_2.64.0              png_0.1-8                  
[101] rstudioapi_0.16.0           reshape2_1.4.4             
[103] tzdb_0.4.0                  rjson_0.2.23               
[105] nlme_3.1-166                curl_5.2.3                 
[107] lolliplot_0.2.2             proxy_0.4-27               
[109] cachem_1.1.0                GlobalOptions_0.1.2        
[111] KernSmooth_2.23-24          parallel_4.4.0             
[113] restfulr_0.0.15             pillar_1.9.0               
[115] vctrs_0.6.5                 promises_1.3.0             
[117] ggpubr_0.6.0                car_3.1-3                  
[119] xtable_1.8-4                evaluate_1.0.0             
[121] mvtnorm_1.3-1               cli_3.6.3                  
[123] compiler_4.4.0              Rsamtools_2.20.0           
[125] rlang_1.1.4                 crayon_1.5.3               
[127] ggsignif_0.6.4              labeling_0.4.3             
[129] classInt_0.4-10             ps_1.8.0                   
[131] plyr_1.8.9                  getPass_0.2-4              
[133] fs_1.6.4                    stringi_1.8.4              
[135] viridisLite_0.4.2           BiocParallel_1.38.0        
[137] munsell_0.5.1               Biostrings_2.72.1          
[139] Matrix_1.7-0                hms_1.1.3                  
[141] shiny_1.9.1                 pegas_1.3                  
[143] SummarizedExperiment_1.34.0 highr_0.11                 
[145] broom_1.0.7                 bslib_0.8.0                
[147] ape_5.8                    
</pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
